ffi/wasm/src/ssh.rs

```diff
@@ -0,0 +1,99 @@
+use crate::pem::Pem;
+use crate::key::{EcCurve, PrivateKey, PublicKey};
+
+use wasm_bindgen::prelude::*;
+
+define_error!(SshPrivateKeyError, picky::ssh::private_key::SshPrivateKeyError);
+define_error!(SshPublicKeyError, picky::ssh::public_key::SshPublicKeyError);
+
+#[wasm_bindgen]
+pub struct SshPrivateKey(pub(crate) picky::ssh::private_key::SshPrivateKey);
+
+#[wasm_bindgen]
+impl SshPrivateKey {
+    /// Generates a new RSA SSH private key
+    pub fn generate_rsa(bits: usize, passphrase: Option<String>, comment: Option<String>) -> Result<SshPrivateKey, SshPrivateKeyError> {
+        let key = picky::ssh::private_key::SshPrivateKey::generate_rsa(bits, passphrase, comment)?;
+        Ok(Self(key))
+    }
+
+    /// Generates a new EC SSH private key
+    pub fn generate_ec(curve: EcCurve, passphrase: Option<String>, comment: Option<String>) -> Result<SshPrivateKey, SshPrivateKeyError> {
+        let key = picky::ssh::private_key::SshPrivateKey::generate_ec(curve.into(), passphrase, comment)?;
+        Ok(Self(key))
+    }
+
+    /// Generates a new Ed25519 SSH private key
+    pub fn generate_ed25519(passphrase: Option<String>, comment: Option<String>) -> Result<SshPrivateKey, SshPrivateKeyError> {
+        let key = picky::ssh::private_key::SshPrivateKey::generate_ed25519(passphrase, comment)?;
+        Ok(Self(key))
+    }
+
+    /// Parses SSH private key from PEM object
+    pub fn from_pem(pem: &Pem, passphrase: Option<String>) -> Result<SshPrivateKey, SshPrivateKeyError> {
+        let key = picky::ssh::private_key::SshPrivateKey::from_pem(&pem.0, passphrase)?;
+        Ok(Self(key))
+    }
+
+    /// Converts SSH private key to PEM object
+    pub fn to_pem(&self) -> Result<Pem, SshPrivateKeyError> {
+        Ok(Pem(self.0.to_pem()?))
+    }
+
+    /// Sets SSH private key comment
+    pub fn set_comment(&mut self, comment: String) {
+        self.0.comment = comment;
+    }
+
+    /// Returns comment of the SSH private key
+    pub fn comment(&self) -> String {
+        self.0.comment.clone()
+    }
+
+    /// Returns cipher name of the SSH private key
+    pub fn cipher_name(&self) -> String {
+        self.0.cipher_name.clone()
+    }
+
+    /// Returns inner private key from SSH private key
+    pub fn inner_key(&self) -> Result<PrivateKey, SshPrivateKeyError> {
+        Ok(PrivateKey(self.0.inner_key()?.clone()))
+    }
+
+    /// Extracts the public part from SSH private key
+    pub fn public_key(&self) -> SshPublicKey {
+        SshPublicKey(self.0.public_key().clone())
+    }
+}
+
+#[wasm_bindgen]
+pub struct SshPublicKey(pub(crate) picky::ssh::public_key::SshPublicKey);
+
+#[wasm_bindgen]
+impl SshPublicKey {
```

Comment:
Added ssh key generation, parsing, encoding as per [ARC-208](https://devolutions.atlassian.net/browse/ARC-208)

Reply:
Should we also add SSH certs in our API or it could be done as a follow-up?

Reply:
I’m fine with a follow-up! :+1:

Reply:
AFAIK, Hub team really needed the API for managing SSH key pairs, but not the certs, so we can reevaluate later.